raclients/src/frontend/mod.rs

```diff
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: Apache-2.0
+
+#[cfg(all(not(feature = "std"), feature = "alloc"))]
+extern crate alloc;
+
+pub(crate) mod lib {
+    pub mod core {
+        #[cfg(all(feature = "std", not(feature = "alloc")))]
+        pub use std::*;
+
+        #[cfg(all(not(feature = "std"), feature = "alloc"))]
+        pub use core::*;
+    }
+
+    pub mod alloc {
+        #[cfg(all(feature = "std", not(feature = "alloc")))]
+        pub use std::*;
+
+        #[cfg(all(not(feature = "std"), feature = "alloc"))]
+        pub use super::super::alloc::*;
+    }
+
+    pub use self::alloc::{
+        format,
+        string::{String, ToString},
+        vec::Vec,
+    };
+}
+
+#[cfg(all(feature = "std", not(feature = "alloc")))]
+pub mod unix;
+
+mod proxy;
+
+pub use proxy::*;
+
+use lib::*;
+
+use serde::{Deserialize, Serialize};
+use serde_json::{Map, Value};
+
+#[derive(Deserialize, Serialize)]
+pub enum NegotiationParam {
+    RsaPubkeyN,
+    RsaPubkeyE,
+    Bytes(Vec<u8>),
+}
+
+pub trait FrontendClient: RacProxyConnection {
```

Comment:
IMHO for a client or server is not really clear the order of the calls.

Please take a look at https://github.com/stefano-garzarella/RAClients/blob/708ff568c15de211f10e58a007adedd253f38c44/src/in_proxy/client_session.rs#L97

We have just 2 calls negotiation() and attestation() where both are req/resp, IMHO is very less error-prone (we also need to add a little state machine there to avoid to call attestation without negotiation).